src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/CoalesceAnalyzer.php

```diff
@@ -44,7 +44,9 @@ public static function analyze(
             || $root_expr instanceof PhpParser\Node\Expr\NullsafeMethodCall
             || $root_expr instanceof PhpParser\Node\Expr\Ternary
         ) {
-            $left_var_id = '$<tmp coalesce var>' . (int) $left_expr->getAttribute('startFilePos');
+            $left_var_id = '$<psalm_coalesce_temp_var>'
+                . (int) $left_expr->getAttribute('startFilePos')
+                . ': array<array-key, mixed>';
```

Comment:
Not sure why `array<array-key, mixed>` is hardcoded here. And it doesn't seem it adds anything useful to the message as `TypeContradiction` would still list the inferred type anyway.

Reply:
The hardcoded type hint array<array-key, mixed> in the variable assignment was initially introduced to provide explicit information about the expected type of the temporary variable. However, after careful consideration and feedback, it appears that this information is already effectively conveyed by TypeContradiction.

To improve code readability and adhere to the feedback received, should I make the necessary changes?

Reply:
yes, please remove it, I think it add confusion instead of reducing it